src/Client.ts

```diff
@@ -34,14 +35,14 @@ export class Client implements IHasHTTP {
    * @param timeout - how long to wait before giving up when making ReST calls
    */
   public static async checkServer(server: OnmsServer, httpImpl?: IOnmsHTTP, timeout?: number): Promise<boolean> {
-    const opts = new OnmsHTTPOptions(timeout, server.auth, server);
+    let opts = new OnmsHTTPOptions(timeout, server);
     if (!httpImpl) {
       if (!Client.defaultHttp) {
         throw new OnmsError('No HTTP implementation is configured!');
       }
       httpImpl = Client.defaultHttp;
     }
-    opts.headers.accept = 'text/plain';
+    opts = opts.withHeader('Accept', 'text/plain');
```

Comment:
Rather than using the `with` methods, could we instead use the builder pattern? If we went that route, we could avoid what is happening here where we created a throwaway copy of `OnmsHTTPOptions` just to then set one of the optional parameters on it, creating a new object in the process.

If we use builder pattern we only ever create 1 builder instance and 1 entity instance no matter how many optional parameters we end up setting. You can also make the reference immutable via `const` rather than using `let` and having to store the intermediary states.

This would then become:
```
const opts = OnmsHTTPOptions.newBuilder()
    .withTimeout(timeout)
    .withServer(server)
    .withHeader('Accept', 'text/plain')
    .build();
```